core/isolate.rs

```diff
@@ -748,8 +763,34 @@ pub mod tests {
     )
   }
 
+  struct DelayedFuture {
+    counter: u32,
+    buf: Box<[u8]>,
+  }
+
+  impl DelayedFuture {
+    pub fn new(buf: Box<[u8]>) -> Self {
+      DelayedFuture { counter: 0, buf }
+    }
+  }
+
+  impl Future for DelayedFuture {
+    type Item = Box<[u8]>;
```

Comment:
Probably could be made generic easily...

Reply:
Of course, but it's used only for a few test cases.

Reply:
yeah I just realized that - ignore my comments